diff --git a/Source/Core/Common/Crypto/AES.cpp b/Source/Core/Common/Crypto/AES.cpp
index a83812ff2e..7d6ddb6463 100644
--- a/Source/Core/Common/Crypto/AES.cpp
+++ b/Source/Core/Common/Crypto/AES.cpp
@@ -4,6 +4,7 @@
 #include "Common/Crypto/AES.h"
 
 #include <array>
+#include <cstring>
 #include <memory>
 
 #include <mbedtls/aes.h>
diff --git a/Source/Core/Common/Logging/Log.h b/Source/Core/Common/Logging/Log.h
index 64d353af00..a8aa89bf23 100644
--- a/Source/Core/Common/Logging/Log.h
+++ b/Source/Core/Common/Logging/Log.h
@@ -96,20 +96,15 @@ static const char LOG_LEVEL_TO_CHAR[7] = "-NEWID";
 void GenericLogFmtImpl(LogLevel level, LogType type, const char* file, int line,
                        fmt::string_view format, const fmt::format_args& args);
 
-template <std::size_t NumFields, typename S, typename... Args>
-void GenericLogFmt(LogLevel level, LogType type, const char* file, int line, const S& format,
-                   const Args&... args)
+template <std::size_t NumFields, typename... Args>
+void GenericLogFmt(LogLevel level, LogType type, const char* file, int line,
+                   fmt::format_string<Args...> format, const Args&... args)
 {
   static_assert(NumFields == sizeof...(args),
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
 
-#if FMT_VERSION >= 110000
-  auto&& format_str = fmt::format_string<Args...>(format);
-#else
-  auto&& format_str = format;
-#endif
-  GenericLogFmtImpl(level, type, file, line, format_str, fmt::make_format_args(args...));
+  GenericLogFmtImpl(level, type, file, line, format.get(), fmt::make_format_args(args...));
 }
 }  // namespace Common::Log
 
diff --git a/Source/Core/Common/MsgHandler.h b/Source/Core/Common/MsgHandler.h
index 4fc7196a97..54429114f7 100644
--- a/Source/Core/Common/MsgHandler.h
+++ b/Source/Core/Common/MsgHandler.h
@@ -32,44 +32,27 @@ void RegisterStringTranslator(StringTranslator translator);
 bool MsgAlertFmtImpl(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
                      int line, fmt::string_view format, const fmt::format_args& args);
 
-template <std::size_t NumFields, typename S, typename... Args>
+template <std::size_t NumFields, typename... Args>
 bool MsgAlertFmt(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
-                 int line, const S& format, const Args&... args)
+                 int line, fmt::format_string<Args...> format, const Args&... args)
 {
   static_assert(NumFields == sizeof...(args),
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
-#if FMT_VERSION >= 110000
-  static_assert(std::is_base_of_v<fmt::detail::compile_string, S>);
-  auto&& format_str = fmt::format_string<Args...>(format);
-#elif FMT_VERSION >= 90000
-  static_assert(fmt::detail::is_compile_string<S>::value);
-  auto&& format_str = format;
-#else
-  static_assert(fmt::is_compile_string<S>::value);
-  auto&& format_str = format;
-#endif
-  return MsgAlertFmtImpl(yes_no, style, log_type, file, line, format_str,
+  return MsgAlertFmtImpl(yes_no, style, log_type, file, line, format.get(),
                          fmt::make_format_args(args...));
 }
 
-template <std::size_t NumFields, bool has_non_positional_args, typename S, typename... Args>
+template <std::size_t NumFields, bool has_non_positional_args, typename... Args>
 bool MsgAlertFmtT(bool yes_no, MsgType style, Common::Log::LogType log_type, const char* file,
-                  int line, const S& format, fmt::string_view translated_format,
-                  const Args&... args)
+                  int line, [[maybe_unused]] fmt::format_string<Args...> format,
+                  fmt::string_view translated_format, const Args&... args)
 {
   static_assert(!has_non_positional_args,
                 "Translatable strings must use positional arguments (e.g. {0} instead of {})");
   static_assert(NumFields == sizeof...(args),
                 "Unexpected number of replacement fields in format string; did you pass too few or "
                 "too many arguments?");
-#if FMT_VERSION >= 110000
-  static_assert(std::is_base_of_v<fmt::detail::compile_string, S>);
-#elif FMT_VERSION >= 90000
-  static_assert(fmt::detail::is_compile_string<S>::value);
-#else
-  static_assert(fmt::is_compile_string<S>::value);
-#endif
   auto arg_list = fmt::make_format_args(args...);
   return MsgAlertFmtImpl(yes_no, style, log_type, file, line, translated_format, arg_list);
 }
diff --git a/Source/Core/Common/TimeUtil.cpp b/Source/Core/Common/TimeUtil.cpp
index 93327e9136..cb03b0bf9d 100644
--- a/Source/Core/Common/TimeUtil.cpp
+++ b/Source/Core/Common/TimeUtil.cpp
@@ -5,6 +5,7 @@
 #include "Common/Logging/Log.h"
 
 #include <ctime>
+#include <cstring>
 #include <optional>
 
 namespace Common
diff --git a/Source/Core/VideoBackends/OGL/OGLBoundingBox.cpp b/Source/Core/VideoBackends/OGL/OGLBoundingBox.cpp
index 9b36a75b63..673e308d53 100644
--- a/Source/Core/VideoBackends/OGL/OGLBoundingBox.cpp
+++ b/Source/Core/VideoBackends/OGL/OGLBoundingBox.cpp
@@ -3,6 +3,8 @@
 
 #include "VideoBackends/OGL/OGLBoundingBox.h"
 
+#include <cstring>
+
 #include "VideoBackends/OGL/OGLGfx.h"
 #include "VideoCommon/DriverDetails.h"
 
