diff --git i/CMakeLists.txt w/CMakeLists.txt
index 6a06ffb..b585a3a 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required (VERSION 3.20)
 
-if(NOT DEFINED ENV{VCPKG_ROOT})
+option(UNLEASHED_RECOMP_USE_SYSTEM_LIBS "Use system libraries" OFF)
+
+if(NOT UNLEASHED_RECOMP_USE_SYSTEM_LIBS AND NOT DEFINED ENV{VCPKG_ROOT})
     message(FATAL_ERROR "VCPKG_ROOT is not defined!")
 endif()
 
@@ -19,6 +21,21 @@ set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
 project("UnleashedRecomp-ALL")
 
+if(UNLEASHED_RECOMP_USE_SYSTEM_LIBS)
+    find_package(PkgConfig REQUIRED)
+    find_package(fmt REQUIRED)
+    find_package(tomlplusplus REQUIRED)
+    find_package(SDL2 REQUIRED CONFIG)
+    find_package(SDL2_mixer REQUIRED CONFIG)
+    pkg_check_modules(xxHash REQUIRED IMPORTED_TARGET libxxhash)
+    add_library(xxHash::xxhash ALIAS PkgConfig::xxHash)
+    pkg_check_modules(zstd REQUIRED IMPORTED_TARGET libzstd)
+    add_library(libzstd ALIAS PkgConfig::zstd)
+    add_library(libzstd_static ALIAS PkgConfig::zstd)
+    add_library(SDL2::SDL2-static ALIAS SDL2::SDL2)
+    add_library(SDL2_mixer ALIAS SDL2_mixer::SDL2_mixer)
+endif()
+
 if (APPLE)
     enable_language(OBJC OBJCXX)
 endif()
@@ -33,13 +50,6 @@ endif()
 string(TOLOWER "${UNLEASHED_RECOMP_ARCHITECTURE}" UNLEASHED_RECOMP_ARCHITECTURE)
 message(STATUS "Detected architecture: ${UNLEASHED_RECOMP_ARCHITECTURE}")
 
-if (UNLEASHED_RECOMP_ARCHITECTURE STREQUAL "x86_64" OR UNLEASHED_RECOMP_ARCHITECTURE STREQUAL "amd64")
-    # Target Sandy Bridge for all projects
-    add_compile_options(
-        -march=sandybridge
-    )
-endif()
-
 add_subdirectory(${UNLEASHED_RECOMP_THIRDPARTY_ROOT})
 add_subdirectory(${UNLEASHED_RECOMP_TOOLS_ROOT})
 
diff --git i/thirdparty/CMakeLists.txt w/thirdparty/CMakeLists.txt
index d5c0858..a0b13ba 100644
--- i/thirdparty/CMakeLists.txt
+++ w/thirdparty/CMakeLists.txt
@@ -23,11 +23,17 @@ if (WIN32)
     set(D3D12_AGILITY_SDK_ENABLED ON CACHE BOOL "")
 endif()
 
+set(MSDF_ATLAS_USE_VCPKG OFF)
+
 add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/msdf-atlas-gen")
 add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/nativefiledialog-extended")
 add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/o1heap")
-add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/SDL")
-add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/SDL_mixer")
+if (NOT TARGET SDL2::SDL2-static)
+    add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/SDL")
+endif()
+if (NOT TARGET SDL2_mixer)
+    add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/SDL_mixer")
+endif()
 add_subdirectory("${UNLEASHED_RECOMP_THIRDPARTY_ROOT}/plume")
 
 if (APPLE)
diff --git i/UnleashedRecomp/stdafx.h w/UnleashedRecomp/stdafx.h
index 4c55391..dff529a 100644
--- i/UnleashedRecomp/stdafx.h
+++ w/UnleashedRecomp/stdafx.h
@@ -45,7 +45,7 @@ using Microsoft::WRL::ComPtr;
 #include <cstddef>
 #include <smolv.h>
 #include <set>
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <list>
 #include <semaphore>
 #include <numeric>
diff --git i/tools/XenonRecomp/XenonAnalyse/main.cpp w/tools/XenonRecomp/XenonAnalyse/main.cpp
index d08371e..f156eab 100644
--- i/tools/XenonRecomp/XenonAnalyse/main.cpp
+++ w/tools/XenonRecomp/XenonAnalyse/main.cpp
@@ -4,7 +4,7 @@
 #include <disasm.h>
 #include <image.h>
 #include <xbox.h>
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include "function.h"
 
 #define SWITCH_ABSOLUTE 0
diff --git i/tools/XenonRecomp/XenonRecomp/pch.h w/tools/XenonRecomp/XenonRecomp/pch.h
index c9f9a85..fa2007f 100644
--- i/tools/XenonRecomp/XenonRecomp/pch.h
+++ w/tools/XenonRecomp/XenonRecomp/pch.h
@@ -15,5 +15,5 @@
 #include <unordered_set>
 #include <xbox.h>
 #include <xxhash.h>
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <x86/sse.h>
diff --git i/tools/XenosRecomp/XenosRecomp/pch.h w/tools/XenosRecomp/XenosRecomp/pch.h
index e59682c..81943ef 100644
--- i/tools/XenosRecomp/XenosRecomp/pch.h
+++ w/tools/XenosRecomp/XenosRecomp/pch.h
@@ -13,7 +13,7 @@
 #include <filesystem>
 #include <map>
 #include <smolv.h>
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <string>
 #include <unordered_map>
 #include <xxhash.h>
